components/select/SelectOption.razor.cs

```diff
@@ -1,169 +1,173 @@
+                    SetClassMap();
+                }
+            }
         }
 
-        #region Properties
+        private bool _isDisabled;
+        public bool IsDisabled
+        {
+            get => _isDisabled;
+            set
+            {
+                if (_isDisabled != value)
+                {
+                    _isDisabled = value;
+                    SetClassMap();
+                }
+            }
+        }
 
-        protected string InnerStyle
+        private bool _isHidden;
+        public bool IsHidden
         {
-            get
+            get => _isHidden;
+            set
             {
-                if (IsSearch)
+                if (_isHidden != value)
                 {
-                    return Style;
+                    _isHidden = value;
+                    SetClassMap();
                 }
+            }
+        }
 
-                if (SelectParent.IsShowOption(this))
+        private bool _isActive;
+        public bool IsActive
+        {
+            get => _isActive;
+            set
+            {
+                if (_isActive != value)
                 {
-                    return Style;
+                    _isActive = value;
+                    SetClassMap();
                 }
+            }
+        }
 
-                return Style + ";display:none";
+        private string _label = string.Empty;
+        public string Label
+        {
+            get => _label;
+            set
+            {
+                if (_label != value)
+                {
+                    _label = value;
+                    StateHasChanged();
+                }
             }
         }
-        #endregion
 
-        #region Events
-        protected override void Dispose(bool disposing)
+        private string _groupName = string.Empty;
+        public string GroupName
         {
-            SelectParent?.RemoveOption(this);
-            SelectOptGroupParent?.RemoveOption(this);
-            base.Dispose(disposing);
+            get => _groupName;
+            set
+            {
+                if (_groupName != value)
+                {
+                    _groupName = value;
+                    StateHasChanged();
+                }
+            }
         }
 
+        #endregion
+
         protected override async Task OnInitializedAsync()
         {
-            SetClassMap();
+            var item = SelectParent.SelectOptions.First(x => x.InternalId == InternalId);
+            item.ChildComponent = this;
 
-            SelectParent?.AddOption(this);
-            SelectOptGroupParent?.AddOption(this);
+            SetClassMap();
 
             await base.OnInitializedAsync();
         }
 
-        protected override void OnInitialized()
+        protected void SetClassMap()
         {
+            ClassMapper.Clear()
+                .Add("ant-select-item")
+                .Add(ClassPrefix)
+                .If($"{ClassPrefix}-disabled", () => IsDisabled)
```

Comment:
We can use the `GetIf` to do the string concatenation each time. And no need to call `SetClassMap()` so many time.
```suggestion
                .GetIf(()=>$"{ClassPrefix}-disabled", () => IsDisabled)
```

Reply:
Hi James,

i tried this: 
```
            ClassMapper.Clear()
                .Add("ant-select-item")
                .Add(ClassPrefix)
                .GetIf(() => $"{ClassPrefix}-disabled", () => IsDisabled)
                .GetIf(() => $"{ClassPrefix}-selected", () => IsSelected)
                .GetIf(() => $"{ClassPrefix}-active", () => IsActive)
                .GetIf(() => $"{ClassPrefix}-grouped", () => SelectParent.IsGroupingEnabled);
```

I need a StateHasChanged after the properties IsSelected, IsDisabled, IsHidden, IsActive. Otherwise there are display issues.

Like this:

```
        public bool IsSelected
        {
            get => _isSelected;
            set
            {
                if (_isSelected != value)
                {
                    _isSelected = value;
                    StateHasChanged();
                }
            }
        }
```

What is the benefit to use GetIf? I can't see the difference. :s